integration/setup/flags.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package setup
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// flags are used to override flags set from cli with test setup option.
+type flags struct {
+	targetPort       *int
+	targetTLS        *bool
+	handler          *string
+	targetUnixSocket *bool
+	proxyAddr        *string
+	compatPort       *int
+	compatTLS        *bool
+	postgreSQLURL    *string
+	tigrisURL        *string
+}
+
+// ApplyOpts applies opts to the flags to override it.
+func (f *flags) ApplyOpts(tb testing.TB, opts map[string]any) *flags {
+	for k, v := range opts {
+		switch k {
+		case "target-port":
+			targetPort, ok := v.(int)
+			require.True(tb, ok, "%s is not int: %T", k, v)
+
+			f.targetPort = &targetPort
+		case "target-tls":
+			targetTLS, ok := v.(bool)
+			require.True(tb, ok, "%s is not bool: %T", v)
+
+			f.targetTLS = &targetTLS
+		case "handler":
+			handler, ok := v.(string)
+			require.True(tb, ok, "%s is not string: %T", v)
+
+			f.handler = &handler
+		case "target-unix-socket":
+			targetUnixSocket, ok := v.(bool)
+			require.True(tb, ok, "%s is not bool: %T", v)
+
+			f.targetUnixSocket = &targetUnixSocket
+		case "proxy-addr":
+			proxyAddr, ok := v.(string)
+			require.True(tb, ok, "%s is not string: %T", v)
+
+			f.proxyAddr = &proxyAddr
+		case "compat-port":
+			compatPort, ok := v.(int)
+			require.True(tb, ok, "%s is not int: %T", k, v)
+
+			f.compatPort = &compatPort
+		case "compat-tls":
+			compatTLS, ok := v.(bool)
+			require.True(tb, ok, "%s is not bool: %T", v)
+
+			f.targetTLS = &compatTLS
```

Comment:
Is this enough as `bool`? Don't we need something like "target-auth-mechanism" and "compat-auth-mechanism" to know how exactly we should connect to target and compat in case of TLS/auth being set? Or is it just for the beginning?

Reply:
First you found bug, it should be assigned to `f.compatTLS = &compatTLS`. I added tests for the flags. The compatTLS and targetTLS should be separate.

For authentication mechanism, it's based on if ferretdb is connecting or mongodb is connecting. see this for my understanding of TLS.
https://github.com/FerretDB/FerretDB/pull/1857#discussion_r1086350806

As far as I know, we share the certificate between target and compat, using flag for it is sufficient. @w84thesun is that right?

Reply:
Yes, we have one set for both MongoDB and FerretDB.